client/src/connection/rest/session.ts

```diff
@@ -177,8 +177,45 @@ export class ComputeSession extends Compute {
   async *getLogStream(options?: {
     timeout?: number;
   }): AsyncGenerator<LogLine[]> {
+    let state = await this.getState();
     const timeout = options?.timeout ?? 10;
-    const start = 0;
+    let start = 0;
+
+    const states = [
+      "done",
+      "canceled",
+      "error",
+      "warning",
+      "completed",
+      "idle",
+    ];
+
+    while (states.indexOf(state) === -1) {
```

Comment:
Is there a chance this ends up an infinite loop?

Reply:
if "idle" were to be removed from the list of states then absolutely. Once the session goes idle, it would cycle there infinitely. Basically the logic we're trying to convey, is that if the session status is pending or not yet "complete", go ahead and grab what log we have and yield it out. In the case where the session spins up quickly on faster connections and becomes "idle" then the logic simply falls down into the block below where we yield out whatever is there and terminate when there are no more pages to yield.